aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -679,21 +921,99 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         let schema_json = serde_json::to_string(&schema)?;
         let schema_id = &schema.id().0;
 
-        // TODO - future - store as cache against issuer_did
+        // TODO - future - store as cache against issuer_did - why? indy_anoncreds.rs doesn't seem to cache it either
```

Comment:
i think i wrote this comment originally. I was most likely using ACA-py as a reference and noticed that they store their `CATEGORY_SCHEMA`/`CATEGORY_CRED_SCHEMA` wallet schema record here. 

Whereas in your implementation, you seem to be storing it in `issuer_create_and_store_credential_def`.

I'm not sure if that is what vdrtools does, but it's a little confusing to me. As an issuer can create many cred defs for the same schema. So wouldn't this throw a duplicate record error if an issuer creates a 2nd cred def for the same schema/schema_id?

Reply:
Also, because this represents a mapping between the cred_def_id and the schema_id, it has to be done when the credential is actually created. Again, this is how libvdrtools does it.

Reply:
Yes IMO that is a valid scenario, and should be possible; I tend to do this workflow at work a lot using ACA-py. 

E.g. creating a generic `University_Degree` schema and then creating several cred defs against that schema for different mock universities `ABC_University_Degree`, `XYZ_University_Degree` etc. 

So this line would not allow for that behaviour, as `schema.id()` might already exist in the wallet if already used for a cred def previously;
```
        self.wallet
            .add_wallet_record(CATEGORY_CRED_SCHEMA, schema.id(), schema_json, None)
            .await?;
```

Reply:
Had a look in libvdrtools, the difference is that they ignore errors on this schema storage, so they are silencing the duplicate wallet record error;

```
        let _ = self
            .wallet_service
            .add_indy_object(wallet_handle, &schema_id.0, &schema_, &HashMap::new())
            .await
            .ok();
```

note `.ok` rather than `?`